src/anndata/experimental/backed/_xarray.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pandas as pd
+
+from ..._core.anndata import AnnData, _gen_dataframe
+from ..._core.file_backing import to_memory
+from ..._core.index import _subset
+from ..._core.views import as_view
+
+try:
+    from xarray import Dataset
+except ImportError:
+
+    class Dataset:
+        def __repr__(self) -> str:
+            return "mock Dataset"
+
+
+if TYPE_CHECKING:
+    from collections.abc import Hashable, Iterable
+    from typing import Any, Literal
+
+    from ..._core.index import Index
+    from ._compat import xarray as xr
+
+
+def get_index_dim(ds: xr.DataArray) -> Hashable:
+    assert (
+        len(ds.sizes) == 1
+    ), f"xarray Dataset should not have more than 1 dims, found {len(ds)}"
+    return list(ds.indexes.keys())[0]
+
+
+class Dataset2D(Dataset):
+    @property
+    def index(self) -> pd.Index:
+        """:attr:`~anndata.AnnData` internally looks for :attr:`~pandas.DataFrame.index` so this ensures usability
+
+        Returns
+        -------
+            The index of the of the dataframe as resolved from :attr:`~xarray.Dataset.coords`.
+        """
+        coord = get_index_dim(self)
+        return self.indexes[coord]
+
+    @index.setter
+    def index(self, val) -> None:
+        coord = get_index_dim(self)
+        self.coords[coord] = val
+
+    @property
+    def shape(self) -> tuple[int, int]:
+        """:attr:`~anndata.AnnData` internally looks for :attr:`~pandas.DataFrame.shape` so this ensures usability
+
+        Returns
+        -------
+            The (2D) shape of the dataframe resolved from :attr:`~xarray.Dataset.sizes`.
+        """
+        return (self.sizes[get_index_dim(self)], len(self))
+
+    @property
+    def iloc(self):
+        """:attr:`~anndata.AnnData` internally looks for :attr:`~pandas.DataFrame.iloc` so this ensures usability
+
+        Returns
+        -------
+            Handler class for doing the iloc-style indexing using :meth:`~xarray.Dataset.isel`.
+        """
+
+        class IlocGetter:
+            def __init__(self, ds):
+                self._ds = ds
+
+            def __getitem__(self, idx):
+                coord = get_index_dim(self._ds)
+                return self._ds.isel(**{coord: idx})
+
+        return IlocGetter(self)
+
+    @property
+    def columns(self) -> pd.Index:
+        """
+        :class:`~anndata.AnnData` internally looks for :attr:`~pandas.DataFrame.columns` so this ensures usability
+
+        Returns
+        -------
+            :class:`pandas.Index` that represents the "columns."
+        """
+        columns_list = list(self.keys())
+        return pd.Index(columns_list)
+
+
+@_subset.register(Dataset2D)
```

Comment:
Do we document this set of functions somewhere? seems like this is the set of functions each new array type needs support for.